test/test-server/server.go

```diff
@@ -50,6 +57,11 @@ type HttpsServerHandler struct {
 }
 
 func (h HttpsServerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if delay, err := strconv.Atoi(r.URL.Query().Get("delay")); err == nil {
+		fmt.Println("delay", delay)
+		time.Sleep(time.Second * time.Duration(delay))
```

Comment:
`strconv.Atoi` use `time.ParseDuration()` that parse string to Duration, that could also support `10s`